### PR TITLE
chore: simplify OptimismGenesisInfo extraction

### LINKED ISSUE #8952: Simplify OptimismGenesis Info extraction
### Describe the feature

with alloy 0.1.2 we can simplify this:

https://github.com/paradigmxyz/reth/blob/590356b3415cffec09f10189ff0ec006d9447d04/crates/chainspec/src/spec.rs#L1398-L1398

and can use the `Otherfields` API to extract all fields:

https://alloy-rs.github.io/alloy/alloy/rpc/types/serde_helpers/struct.OtherFields.html#method.deserialize_as

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Vid201**
Closes #8952

### CODE REVIEW

File: crates/chainspec/src/spec.rs [hunk 1]
```diff
@@ -26,6 +26,8 @@ use reth_primitives_traits::{
     Header, SealedHeader,
 };
 use reth_trie_common::root::state_root_ref_unhashed;
+#[cfg(feature = "optimism")]
+use serde::Deserialize;
```
Comment (mattsse) [inline crates/chainspec/src/spec.rs:30]: please use fullpath instead of feature gating the import

**Review by mattsse — CHANGES_REQUESTED** [review]
ty,

can we also add a simple test case for this:

```rust
  "config": {
    "chainId": 8453,
    "homesteadBlock": 0,
    "eip150Block": 0,
    "eip155Block": 0,
    "eip158Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "petersburgBlock": 0,
    "istanbulBlock": 0,
    "muirGlacierBlock": 0,
    "berlinBlock": 0,
    "londonBlock": 0,
    "arrowGlacierBlock": 0,
    "grayGlacierBlock": 0,
    "mergeNetsplitBlock": 0,
    "bedrockBlock": 0,
    "regolithTime": 0,
    "terminalTotalDifficulty": 0,
    "terminalTotalDifficultyPassed": true,
    "optimism": {
      "eip1559Elasticity": 6,
      "eip1559Denominator": 50
    }
  },
```

**Comment by Vid201** [comment]
> ty,
> 
> can we also add a simple test case for this:
> 
> ```rust
>   "config": {
>     "chainId": 8453,
>     "homesteadBlock": 0,
>     "eip150Block": 0,
>     "eip155Block": 0,
>     "eip158Block": 0,
>     "byzantiumBlock": 0,
>     "constantinopleBlock": 0,
>     "petersburgBlock": 0,
>     "istanbulBlock": 0,
>     "muirGlacierBlock": 0,
>     "berlinBlock": 0,
>     "londonBlock": 0,
>     "arrowGlacierBlock": 0,
>     "grayGlacierBlock": 0,
>     "mergeNetsplitBlock": 0,
>     "bedrockBlock": 0,
>     "regolithTime": 0,
>     "terminalTotalDifficulty": 0,
>     "terminalTotalDifficultyPassed": true,
>     "optimism": {
>       "eip1559Elasticity": 6,
>       "eip1559Denominator": 50
>     }
>   },
> ```

added

**Review by mattsse — APPROVED** [review]
ty!

**Pull request merged**